src/explorer/CosmosDBTreeItem.ts

```diff
@@ -43,72 +49,63 @@ export class CosmosDBTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
-                if (databaseTreeItem) {
-                    const label = CosmosDBConnection.makeLabel(databaseTreeItem);
-                    if (!usedLabels[label]) {
-                        usedLabels[label] = true;
-                        return new CosmosDBConnection(this, databaseTreeItem, key);
-                    }
-                }
-                return undefined;
+            async (c: IDetectedConnection) => {
+                const databaseTreeItem = await ext.cosmosAPI.findTreeItem({
+                    connectionString: c.connectionString
+                });
+                return databaseTreeItem ? new CosmosDBConnection(this, databaseTreeItem, c.keys) : undefined;
             },
-            (key: string) => key
+            (c: IDetectedConnection) => c.keys[0] // just use first key for label if connection is invalid
         );
 
         if (treeItems.length > 0) {
             return treeItems;
+        } else {
+            return [new GenericTreeItem(this, {
+                commandId: 'appService.AddCosmosDBConnection',
+                contextValue: 'AddCosmosDBConnection',
+                label: 'Add Cosmos DB Connection...'
+            })];
         }
-
-        return [new GenericTreeItem(this, {
-            commandId: 'appService.AddCosmosDBConnection',
-            contextValue: 'AddCosmosDBConnection',
-            label: 'Add Cosmos DB Connection...'
-        })];
     }
 
     public async createChildImpl(showCreatingTreeItem: (label: string) => void): Promise<AzureTreeItem<ISiteTreeRoot>> {
         const databaseToAdd = await ext.cosmosAPI.pickTreeItem({
-            resourceType: 'Database',
-            apiType: ['Mongo']
+            resourceType: 'Database'
         });
         if (!databaseToAdd) {
             throw new UserCancelledError();
         }
-        const appSettings = await this.root.client.listApplicationSettings();
-        const appSettingKeyToAdd: string = await ext.ui.showInputBox({
-            prompt: 'Enter new connection setting key',
-            validateInput: (v?: string): string | undefined => validateAppSettingKey(appSettings, v),
-            value: "MONGO_URL"
-        });
+        const appSettingsDict = await this.root.client.listApplicationSettings();
         // tslint:disable-next-line:strict-boolean-expressions
-        appSettings.properties = appSettings.properties || {};
-        appSettings.properties[appSettingKeyToAdd] = databaseToAdd.connectionString;
-        await this.root.client.updateApplicationSettings(appSettings);
+        appSettingsDict.properties = appSettingsDict.properties || {};
+
+        const newAppSettings: Map<string, string> = databaseToAdd.docDBData ?
+            await this.promptForDocDBAppSettings(appSettingsDict, databaseToAdd) :
+            await this.promptForMongoAppSettings(appSettingsDict, databaseToAdd);
+
+        for (const [k, v] of newAppSettings) {
+            appSettingsDict.properties[k] = v;
```

Comment:
Do we need to alert users if they're overwriting an existing setting?

Reply:
`promptForMongoAppSettings` and `promptForDocDBAppSettings ` use `validateInput` to prevent them from getting in that situation